apps/api/src/app/workflows-v2/mappers/workflow-template-get-mapper.ts

```diff
@@ -0,0 +1,77 @@
+import { IPreferenceChannels, StepTypeEnum, WorkflowOriginEnum } from '@novu/shared';
+import { DiscoverWorkflowOutputPreferences } from '@novu/framework';
+import { NotificationStepEntity, NotificationTemplateEntity } from '@novu/dal';
+
+import { MinifiedResponseWorkflowDto, StepResponseDto, WorkflowResponseDto } from '../dto/workflow.dto';
+
+export class WorkflowTemplateGetMapper {
+  static toResponseWorkflowDto(template: NotificationTemplateEntity): WorkflowResponseDto {
+    return {
+      _id: template._id,
+      tags: template.tags,
+      active: template.active,
+      critical: template.critical,
+      notificationGroupId: template._notificationGroupId,
+      preferences: WorkflowTemplateGetMapper.toPreferences(template.preferenceSettings), // Assuming this directly maps; may need adjustment
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** Is this comment out of date now?‏

Reply:
will remove comment